### ISSUE TITLE
NullPointerException in browseSearchHandler

### ISSUE THREAD

**Issue by SubJunk** [UniversalMediaServer/UniversalMediaServer#2709]
I got this from some logs for version 10.10.1 from https://www.universalmediaserver.com/forum/viewtopic.php?f=4&t=14809
Not sure if it has been fixed since then but I wanted to make a record of it in case it hasn't been. Any ideas @ik666 ?

```
DEBUG 2021-09-05 12:22:50.636 [HTTPv2 Request Worker 2] net.pms.network.RequestHandlerV2 Caught exception: null
TRACE 2021-09-05 12:22:50.637 [HTTPv2 Request Worker 2] net.pms.network.RequestHandlerV2 
java.lang.NullPointerException: null
	at net.pms.network.RequestV2.browseSearchHandler(RequestV2.java:1176)
	at net.pms.network.RequestV2.browseHandler(RequestV2.java:1105)
	at net.pms.network.RequestV2.answer(RequestV2.java:687)
	at net.pms.network.RequestHandlerV2.writeResponse(RequestHandlerV2.java:414)
	at net.pms.network.RequestHandlerV2.messageReceived(RequestHandlerV2.java:270)
	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.handleUpstream(SimpleChannelUpstreamHandler.java:70)
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564)
	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendUpstream(DefaultChannelPipeline.java:791)
	at org.jboss.netty.handler.stream.ChunkedWriteHandler.handleUpstream(ChunkedWriteHandler.java:142)
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564)
	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendUpstream(DefaultChannelPipeline.java:791)
	at org.jboss.netty.handler.codec.http.HttpChunkAggregator.messageReceived(HttpChunkAggregator.java:145)
	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.handleUpstream(SimpleChannelUpstreamHandler.java:70)
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564)
	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendUpstream(DefaultChannelPipeline.java:791)
	at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:296)
	at org.jboss.netty.handler.codec.frame.FrameDecoder.unfoldAndFireMessageReceived(FrameDecoder.java:459)
	at org.jboss.netty.handler.codec.replay.ReplayingDecoder.callDecode(ReplayingDecoder.java:536)
	at org.jboss.netty.handler.codec.replay.ReplayingDecoder.messageReceived(ReplayingDecoder.java:435)
	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.handleUpstream(SimpleChannelUpstreamHandler.java:70)
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564)
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:559)
	at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:268)
	at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:255)
	at org.jboss.netty.channel.socket.nio.NioWorker.read(NioWorker.java:88)
	at org.jboss.netty.channel.socket.nio.AbstractNioWorker.process(AbstractNioWorker.java:108)
	at org.jboss.netty.channel.socket.nio.AbstractNioSelector.run(AbstractNioSelector.java:337)
	at org.jboss.netty.channel.socket.nio.AbstractNioWorker.run(AbstractNioWorker.java:89)
	at org.jboss.netty.channel.socket.nio.NioWorker.run(NioWorker.java:178)
	at org.jboss.netty.util.ThreadRenamingRunnable.run(ThreadRenamingRunnable.java:108)
	at org.jboss.netty.util.internal.DeadLockProofWorker$1.run(DeadLockProofWorker.java:42)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by ik666** [comment]
Hey SubJunk, this is actually old code ... The 10.10.1 branch line 1176 is this one:

```

List<DLNAResource> files = PMS.get().getRootFolder(mediaRenderer).getDLNAResources(
			objectID,
			browseDirectChildren,
			startingIndex,
			requestCount,
			mediaRenderer,
			searchCriteria
		);
```

I believe the NPE comes from the call to `getRootFolder(mediaRenderer)` delivering a NULL object, because there is no default renderer available. The board discussion mentions, the renderer is not identified correctly. 

In detail : The `getRootFolder` implementation returns NULL in case the `RendererConfiguration.getDefaultConf()` is null, leading to a NPE in the above code: 

```
	public RootFolder getRootFolder(RendererConfiguration renderer) {
		// something to do here for multiple directories views for each renderer
		if (renderer == null) {
			renderer = RendererConfiguration.getDefaultConf();
		}

		if (renderer == null) {
			return null;
		}

		return renderer.getRootFolder();
	}
```

I think this issue disappears, in case a default renderer is configured.

**Comment by ik666** [comment]
I'd suggest to add a log-warning in case NULL is returned, so the user is aware of this issue, by looking at the logs.

**Comment by ik666** [comment]
Strange though: it's hard to set the defaultRenderer to NULL ... and there are no changes between 10.7 and 10.13 ... I think some more excessive logging is needed ...

**Comment by SubJunk** [comment]
Yeah I had a look and thought the same thing, I'm not sure what could have changed there to cause it. More logging sounds good

**Comment by ik666** [comment]
I think I got it. It came from class BrowseHandler around line 393+-  : `dlna = root.getDLNAResource(id, null);` The passed renderer is null, leading to NPE. Probably fixed this "en passant" with PR #2822.

**Comment by SubJunk** [comment]
Fixed in #3035
